src/scalar_impl.h

```diff
@@ -71,6 +74,14 @@ static void secp256k1_scalar_split_lambda_verify(const secp256k1_scalar *r1, con
 #endif
 
 /*
+ * The function below splits k into r1 and r2, such that
+ * - r1 + lambda * r2 == k (mod n)
+ * - either r1 < 2^128 or -r1 mod n < 2^128
+ * - either r2 < 2^128 or -r2 mod n < 2^128
+ *
+ * It is required that `r1`, `r2`, and `k` all point to different objects. An
```

Comment:
nit: I think this should be documented in `scalar.h` instead.

Reply:
Thanks. I totally missed the declaration there. Fixed.